src/t_zset.c

```diff
@@ -1173,6 +1228,13 @@ unsigned char *zzlDeleteRangeByRank(unsigned char *zl, unsigned int start, unsig
  * Common sorted set API
  *----------------------------------------------------------------------------*/
 
+/* Utility function used for mapping the hashtable entry to the matching skiplist node.
+ * For example, this is used in case of ZRANK query. */
+static inline zskiplistNode *zsetGetSLNodeByEntry(dictEntry *de) {
+    double *score_ref = ((double *)dictGetVal(de));
+    return (zskiplistNode *)(score_ref - 1);
+}
```

Comment:
Oooh, exciting stuff... So the dict value is a pointer to the double score within the skiplist node?

This becomes quite weird here. A better design would be that the dict value points to the skiplist node. Would that be a very big change?

Reply:
I also thought about it but wanted to keep this change modest. Since we will probably change the structure after the hashtable introduction I think making a bigger change is not very important right now.

Reply:
I wonder why this works though and if it works on all possible platforms and compilers. The struct is

```C
typedef struct zskiplistNode {
    sds ele;
    double score;
    ...
} zskiplistNode;
```

We have a pointer to `score`. We treat it as an array of doubles and look at the previous element, which is `ele`, but `ele` is a pointer. On 32-bit, a pointer and a double are not of the same size, but due to alignment, it happens to work anyway.

Should we just trust the tests here then? We do try to support some odd system where we don't have CI. This definitely needs a follow-up...

Reply:
Absolutly true. will fix it now